sd/StringParser/src/com/company/StringFilter.java

```diff
@@ -0,0 +1,33 @@
+package com.company;
+import java.util.*;
+
+public class StringFilter {
+
+    public static String filter(String inputText, HashSet<String> blackList)
+    {
+        boolean firstPass = true;
+        Scanner scanner = new Scanner(inputText);
+        StringBuilder stringBuilder = new StringBuilder();
+
+        // \s is a Java Pattern that represents all forms of white-space
+        scanner.useDelimiter("\\s");
+
+        while(scanner.hasNext())
+        {
+            String currentWord = scanner.next();
+            if (blackList.contains(currentWord.toUpperCase()))
```

Comment:
what if blacklist word or input word has trailing spaces?

Reply:
Good thought, this is solved by the delimiter thankfully